src/tempstorage/TemporaryStorage.py

```diff
@@ -124,11 +124,20 @@ def getSize(self):
     def _clear_temp(self):
         now = time.time()
         if now > (self._last_cache_gc + self._conflict_cache_gcevery):
-            temp_cc = self._conflict_cache.copy()
-            for k, v in temp_cc.items():
-                data, t = v
-                if now > (t + self._conflict_cache_maxage):
-                    del self._conflict_cache[k]
+            # build {} oid -> [](serial, data, t)
+            byoid = {}
+            for ((oid,serial), (data,t)) in self._conflict_cache.items():
+                hist = byoid.setdefault(oid, [])
+                hist.append((serial, data, t))
+
+            # gc entries but keep latest record for each oid
+            for oid, hist in byoid.items():
+                hist.sort(key=lambda _: _[0]) # by serial
```

Comment:
You could use `heapq.nlargest` here (as your only interest is to determine the highest serial) -- not sure whether it is important, though.

Reply:
Thanks. I would prefer to keep this code in the most simplest form for now. If we really see that this place starts to show in profiles or something like that, then it would be better to redo TemporaryStorage data structures and just keep all oid entries in rev↑ order in some single place, instead of `._index` + `.__conflict_cache`.